server/template.go

```diff
@@ -11,8 +11,10 @@ import (
 )
 
 const mdCommentRegexPattern string = `(<!--[\S\s]+?-->)`
+const gitHubRegexPattern string = `(^|[^_\x60[:alnum:]])(@[[:alnum:]](-?[[:alnum:]]+)*)`
```

Comment:
Thank you for the detailed breakdown in the PR description. Can you also link out to the definition of what constitutes a valid GitHub username, should we need to refer back to that any point?

Reply:
I note that this might match `test+@agarciamontoro.com`, which is otherwise a valid email address, but I suspect this would be rare compared to the flexibility of matching `let's ask @lieut-data+@agarciamontoro`, so no changes needed.

Reply:
> Can you also link out to the definition of what constitutes a valid GitHub username, should we need to refer back to that any point?

There is no public definition of what constitutes a valid GitHub username. What I have used here are the error messages returned in https://github.com/join (inspired by https://github.com/shinnn/github-username-regex); namely:
  1. The username must be between 1 and 39 characters long.
  2. The username may only contain alphanumeric characters or single hyphens (i.e., not two adjacent hyphens, as in `@wrong--username`).
  3. The username cannot begin or end with a hyphen.

I can add that as a comment before the `gitHubRegexPattern` definition.

> I note that this might match test+@agarciamontoro.com

This matches indeed test+@agarciamontoro, but as you can see in this same comment, GitHub matches it as a username. I think preserving what GitHub does here is the best thing.

Reply:
Done in https://github.com/mattermost/mattermost-plugin-github/pull/177/commits/c43784aa31f7a9352b578750d82c8b4999347dc2.